src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponent.cs

```diff
@@ -99,13 +102,60 @@ public ModelStateDictionary ModelState
         /// Gets or sets the <see cref="ViewContext"/>.
         /// </summary>
         [Activate]
-        public ViewContext ViewContext { get; set; }
+        public ViewContext ViewContext
+        {
+            get
+            {
+                if (_viewContext == null)
+                {
+                    // This should run only for the ViewComponent unit test scenarios
+                    _viewContext = new ViewContext();
+                    _viewContext.ViewData = new ViewDataDictionary(
+                        new EmptyModelMetadataProvider(),
+                        _viewContext.ModelState);
+                }
+
+                return _viewContext;
+            }
+
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _viewContext = value;
+            }
+        }
 
         /// <summary>
         /// Gets or sets the <see cref="ViewDataDictionary"/>.
         /// </summary>
         [Activate]
-        public ViewDataDictionary ViewData { get; set; }
+        public ViewDataDictionary ViewData
+        {
+            get
+            {
+                if (_viewData == null)
+                {
+                    // This should run only for the ViewComponent unit test scenarios
+                    _viewData = new ViewDataDictionary(ViewContext.ViewData);
```

Comment:
Thanks @dougbu 

The reason why I created a new instance here because it seems like the default behavior for viewcomponent from the following code: 
https://github.com/aspnet/Mvc/blob/2b82ee0255e223ce2347e0e5b7067c7140779db5/src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentActivator.cs#L52-L65
And I think we want to took a snap shot for viewdata for viewcomponent. 

@rynowak, could you clarify which the correct behavior is..